### ISSUE TITLE
Keep rolledback state on agent after rollback is successful

### ISSUE THREAD

**Issue by juliaElastic** [elastic/elastic-agent#6283]
Related to https://github.com/elastic/ingest-dev/issues/2878

With the automatic upgrades feature, we want to ensure that agents that try to upgrade and are rolled back multiple times can be identified and flagged to the users. In order to handle these rollbacks as retry attempts, we need to ensure that the agent keeps `UPG_ROLLBACK` state in `upgrade_details` after rollback is successful.

This is a non-blocker enhancement to the automatic upgrades feature.

**Comment by ycombinator** [comment]
> we need to ensure that the agent keeps `UPG_ROLLBACK` state in `upgrade_details` after rollback is successful.

Consider the following sequence of events:
1. User attempts to upgrade Agent from version X to Y.
2. Upgrade fails and Agent is rolled back to version X.
3. The user or system (due to automatic retries) once again attempts to upgrade Agent from version X to Y.
4. Upgrade fails and Agent is rolled back to version X.
5. The user or system (due to automatic retries) once again attempts to upgrade Agent from version X to Y.
6. Upgrade succeeds and Agent is now on version Y.

In this sequence, at the end of step 4, do we just want to know that the upgrade has been rolled back (regardless of how many times it has been rolled back) or do we want to know _how many times_ the upgrade has been rolled back or do we want to keep a _list of the times_ the upgrade was rolled back?

**Comment by juliaElastic** [comment]
Yes, the idea is that the upgrade retry attempts will be increased when the agent was previously in rolled back state.
This is captured in [this issue](https://github.com/elastic/ingest-dev/issues/4720) (the logic will be part of the async task in kibana fleet plugin).
